diff -rupN cminpack-1.1.4/CMakeLists.txt cminpack-1.1.4.cotire/CMakeLists.txt
--- cminpack-1.1.4/CMakeLists.txt	2011-04-15 08:51:13.000000000 +0200
+++ cminpack-1.1.4.cotire/CMakeLists.txt	2012-03-24 20:21:19.000000000 +0100
@@ -6,6 +6,7 @@ project (CMINPACK)
 string(TOLOWER ${PROJECT_NAME} PROJECT_NAME_LOWER)
 
 include(${PROJECT_SOURCE_DIR}/cmake/cminpack_utils.cmake)
+include(${PROJECT_SOURCE_DIR}/cmake/cotire.cmake)
 # Set version and OS-specific settings
 set(CMINPACK_VERSION 1.0.90 CACHE STRING "CMinpack version")
 DISSECT_VERSION()
@@ -70,3 +71,6 @@ endif (USE_FPIC AND NOT SHARED_LIBS)
 
 set_target_properties(cminpack PROPERTIES VERSION ${CMINPACK_VERSION})
 
+if (COMMAND cotire)
+  cotire(cminpack)
+endif()
